mlflow/utils/model_utils.py

```diff
@@ -34,12 +34,17 @@ def _get_flavor_configuration(model_path, flavor_name):
         )
 
     model_conf = Model.load(model_configuration_path)
-    if flavor_name not in model_conf.flavors:
-        raise MlflowException(
-            'Model does not have the "{flavor_name}" flavor'.format(flavor_name=flavor_name),
-            RESOURCE_DOES_NOT_EXIST,
-        )
-    conf = model_conf.flavors[flavor_name]
+    if flavor_name == "tensorflow" and "keras" in model_conf.flavors:
+        # Patch code for backwards compatibility on loading keras model saved by old
+        # mlflow version.
+        conf = model_conf.flavors["keras"]
+    else:
+        if flavor_name not in model_conf.flavors:
+            raise MlflowException(
+                'Model does not have the "{flavor_name}" flavor'.format(flavor_name=flavor_name),
+                RESOURCE_DOES_NOT_EXIST,
+            )
+        conf = model_conf.flavors[flavor_name]
```

Comment:
Instead of modifying `_get_flavor_configuration()`, can we change the calling code in `mlflow.tensorflow` to call `_get_flavor_configuration(model_path, "tensorflow")` and catch `RESOURCE_DOES_NOT_EXIST` Exceptions? If a `RESOURCE_DOES_NOT_EXIST` exception occurs, we should then call `_get_flavor_configuration(model_path, "keras"). If that fails, we should raise an exception saying that neither the TensorFlow flavor nor the Keras flavor was found.

Reply:
good idea!